File: ts/7.ts
function productExceptSelf(arr: number[]): number[] {
  const results: number[] = [];

  let fullProduct = 1;

  for (const num of arr) {
    fullProduct *= num;
  }

  for (const num of arr) {
    results.push(fullProduct / num);
  }

  return results;
}

console.log(productExceptSelf([1,2,4,6]));

File: python/7.py
def top_k_elements(arr, k):
    frequencies = dict()

    for i in arr:
        if i not in frequencies:
            frequencies[i] = 1
            continue
        frequencies[i] += 1

    sorted_freq = list(reversed(sorted(frequencies.items(), key=lambda x: x[1])))
    results = []
    
    for number, _ in sorted_freq:
        results.append(number)

    return results[:k]

print(top_k_elements([1,2,2,3,3,3], 2))